parsing/cLexer.ml

```diff
@@ -723,7 +723,11 @@ let rec next_token ~diff_mode ttree loc s =
       Stream.junk () s;
       let ep = Stream.count s in
       let t,new_between_commands =
-        if !between_commands then (KEYWORD (String.make 1 c), set_loc_pos loc bp ep), true
+        if !between_commands then begin
+          (* peek to force reading a following newline, cf #19355 *)
+          let _ = Stream.peek () s in
```

Comment:
Does it work if the user wrote something before the newline, e.g., a whitespace?

Reply:
no
same as `.`

Reply:
What will `Stream.peek` do if } is the last token on the stream?

Reply:
It returns None.

Said that the code calls `input`, instead of `really_input`, so it looks fishy

Reply:
I couldn't come up with a case where this would be a problem, but indeed I'm not sure doing the peek here is a good idea; would be good if that could be moved to coqtop.

Reply:
> What will this do if } is the last token on the stream?

Answering to my own question, it will always return `None`.

Reply:
> same as `.`

Ouch, I did not know location reporting was so broken. I just tested it and it is indeed quite impressive.

Reply:
pushed version which does the peek in coqloop resynch_buffer and fixes the location when the command is followed by spaces